### PR TITLE
SON200 - SON Down proposal broken after latest merges

### PR DESCRIPTION
**Pull request by satyakoneru**
SON Down proposal is broken after latest merges, proper account auth not sent in proposals.

### CODE REVIEW

File: libraries/chain/db_maint.cpp [hunk 1]
```diff
@@ -465,9 +465,12 @@ void database::update_active_sons()
       else
       {
          vote_counter vc;
+         a.owner.account_auths.clear();
```
Comment (obucinac) [inline libraries/chain/db_maint.cpp]: Also update owner authority in if branch

Comment (satyakoneru) [reply libraries/chain/db_maint.cpp]: done.

**Comment by satyakoneru** [comment]
> Please verify that setting up owner authority is not preventing us from updating son-account in case when we do not have enough signatures. eg, execute delete_son for half of the sons

That's the requirement, we have to have minimum required signatures to update son-account, we aren't able to update the account without them as expected.

**Comment by obucinac** [comment]
> > Please verify that setting up owner authority is not preventing us from updating son-account in case when we do not have enough signatures. eg, execute delete_son for half of the sons
> 
> That's the requirement, we have to have minimum required signatures to update son-account, we aren't able to update the account without them as expected.

We are able to update active _authority with empty owner authority. We need to know are we able to update active authority once the ownership of account is changed

**Comment by satyakoneru** [comment]
> > > Please verify that setting up owner authority is not preventing us from updating son-account in case when we do not have enough signatures. eg, execute delete_son for half of the sons
> > 
> > 
> > That's the requirement, we have to have minimum required signatures to update son-account, we aren't able to update the account without them as expected.
> 
> We are able to update active _authority with empty owner authority. We need to know are we able to update active authority once the ownership of account is changed

Added the owner weight threshold to the SON account similar to witness and committee account.
It works as expected, at least 2/3 SON active approvals are required for proposals to be successful.
And when the signatures are not present i.e. active weight threshold is not met, proposals don't go through.

**Pull request merged**